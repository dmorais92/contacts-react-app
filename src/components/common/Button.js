import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import cx from "../../utils/classnames";

import "./Button.scss";

class Button extends PureComponent {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.string,
      PropTypes.element,
      PropTypes.array
    ]),
    classes: PropTypes.array,
  };

  static defaultProps = {
    children: null,
    classes: [],
  };

  render() {
    const { children, classes, ...otherProps } = this.props;
    const classNames = [...classes];

    return <button className={`Button${cx(classNames)}`} {...otherProps}>{children}</button>;
  }
}

export default Button;
