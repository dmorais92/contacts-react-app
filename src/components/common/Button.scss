@import '../../variables.scss';

.Button {
  border: 1px solid $default-font-color;
  border-radius: 4px;
  padding: 5px;
  box-shadow: $button-box-shadow;
  background: #fff;
  &:hover {
    box-shadow: $button-box-shadow-hover;
    cursor: pointer;
  }
  &:focus {
    border: 2px solid $primary-border-color
  }
}
