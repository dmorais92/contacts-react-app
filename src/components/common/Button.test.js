import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Card';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders multiple children', () => {
  const component = renderer.create(
    <Button>
      <h1>Testing with multiple children</h1>
      A short paragraph
    </Button>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with extra classes', () => {
  const component = renderer.create(
    <Button classes={["an", "array", "of classes"]}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});