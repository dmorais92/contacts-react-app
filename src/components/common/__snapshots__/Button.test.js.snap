// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiple children 1`] = `
<button
  className="Button"
>
  <h1>
    Testing with multiple children
  </h1>
  A short paragraph
</button>
`;

exports[`renders with extra classes 1`] = `
<button
  className="Button an array of classes"
/>
`;
